hal/armv7m/stm32/l4/stm32l4.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * STM32L4x6 basic peripherals control functions
+ *
+ * Copyright 2017, 2020, 2021 Phoenix Systems
+ * Author: Aleksander Kaminski, Pawel Pisarczyk, Hubert Buczynski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _HAL_STM32L4_H_
+#define _HAL_STM32L4_H_
```

Comment:
I would like to ask You to clean the `stm32l1.h` and `stm32l4.c`. The common part for `Cortex-M` has been divided and it would be great to use such generic functions. I know that You probably copied the code form kernel and it is the reason why this module has some extra code.

The generic functions for `Cortex-M` can be found: https://github.com/phoenix-rtos/plo/tree/feature/syspage/hal/armv7m

Reply:
Should be done